### PR TITLE
Use updated media indicator in Story Promo and Index Alsos

### LINKED ISSUE #3039: Update Media Indicator in Index Alsos
**Is your feature request related to a problem? Please describe.**
After merging https://github.com/bbc/psammead/pull/3029 we need to update how we inline the media indicator in Index Alsos

**Describe the solution you'd like**
Update [Index Alsos](https://github.com/bbc/simorgh/blob/latest/src/app/containers/StoryPromo/IndexAlsos/index.jsx#L33) to pass in `isInline` prop.

Add tests for inline media indicator.

**Testing notes**
Manual checking of storybook should be carried out

Dev insight: 
- [The media indicator example on index alsos should remain the same (only removing the margin that's on the svg)](https://bbc.github.io/psammead/?path=/story/components-storypromo-storypromo--top-story-promo-index-alsos-multiple). 

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by sareh**
Resolves #3039

**Overall change:** Use updated media indicator in Story Promo Index Alsos
Update stories in Psammead Grid, Psammead Story Promo to use updated major version of Media Indicator `@bbc/psammead-media-indicator@4`

**[Storybook Story Promo with Audio story](http://localhost:8180/iframe.html?id=components-storypromo-storypromo--top-story-promo&knob-Select%20a%20service=news&knob-Media%20Type=audio&knob-Image%20alt%20text=Robert%20Downey%20Junior%20in%20Iron%20Man&knob-Image%20src=https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg)**
Screen Shot 2020-02-07 at 08 20 38

**[Storybook Regular Promo No image with Video](http://localhost:8180/iframe.html?id=components-storypromo-storypromo--regular-promo-no-image&knob-Select%20a%20service=news&knob-Media%20Type=video&knob-Image%20alt%20text=Robert%20Downey%20Junior%20in%20Iron%20Man&knob-Image%20src=https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg)**
Screen Shot 2020-02-07 at 08 50 57

**[Grid example with Story Promos with Media indicators](http://localhost:8180/?path=/story/components-grid--example-with-top-story-and-regular-promos)**

Note: the lack of padding here will only be fixed once the psammead-story-promo package update here is pulled into top-level psammead as a dependency. 
Screen Shot 2020-02-07 at 08 49 52

**Code changes:**

- Update to use  `@bbc/psammead-media-indicator@4.0.0`
- Using the `isInline` prop to switch between block and inline versions
- Ensuring headlines etc that are inline with the media indicator have the `display: inline` property so they wrap around the media indicator and below the icon, as per designs. 

**Testing notes**
- [The media indicator example on index alsos should remain the same (only removing the margin that's on the svg)](https://bbc.github.io/psammead/?path=/story/components-storypromo-storypromo--top-story-promo-index-alsos-multiple). 
- [In this case - storypromo--regular-promo-no-image](https://bbc.github.io/psammead/iframe.html?id=components-storypromo-storypromo--regular-promo-no-image&knob-Select%20a%20service=news&knob-Media%20Type=audio&knob-Image%20alt%20text=Robert%20Downey%20Junior%20in%20Iron%20Man&knob-Image%20src=https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg) the media indicator and headline should be in alignment. 
- [In this case - storypromo--regular-promo with image and media indicator, there should be no change](https://bbc.github.io/psammead/iframe.html?id=components-storypromo-storypromo--regular-promo&knob-Select%20a%20service=news&knob-Media%20Type=video&knob-Image%20alt%20text=Robert%20Downey%20Junior%20in%20Iron%20Man&knob-Image%20src=https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg)
---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 2.0.73 | [PR#3073](https://github.com/bbc/psammead/pull/3073) Update `@bbc/psammead-media-indicator` to major version 4 & update relvant usage in packages. |
```
Comment (EinsteinNjoroge) [inline CHANGELOG.md]:
```suggestion
| 2.0.73 | [PR#3073](https://github.com/bbc/psammead/pull/3073) Update `@bbc/psammead-media-indicator` to major version 4 & update relevant usage in packages. |
```

File: packages/components/psammead-grid/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 1.1.7 | [PR#3073](https://github.com/bbc/psammead/pull/3073) Update Grid Stories to include new major version of MediaIndicator |
```
Comment (DenisHdz) [inline packages/components/psammead-grid/CHANGELOG.md:6]: After this PR gets merged, we have to bump the `psammead-story-promo` in the root package.json, to make sure these changes are applied correctly.

File: packages/components/psammead-story-promo/src/index.test.jsx [hunk 3]
```diff
@@ -75,9 +77,14 @@ Info.defaultProps = {
   dir: 'ltr',
   promoHasImage: true,
 };
-
+const TimeDuration = styled.time`
```
Comment (DenisHdz) [inline packages/components/psammead-story-promo/src/index.test.jsx]: Shall this be like the one defined in the stories https://github.com/bbc/psammead/pull/3073/files#diff-02bd1c67c83bcf7bc0fac79c4d66fb3bR27?

File: packages/components/psammead-story-promo/src/index.test.jsx [hunk 4]
```diff
@@ -75,9 +77,13 @@ Info.defaultProps = {
...
   promoHasImage: true,
 };
-
+const StyledTime = styled.time`
+  padding: 0 ${GEL_SPACING_HLF};
+`;
 const mediaInfo = (
-  <MediaIndicator duration="2:15" datetime="PT2M15S" service="news" />
+  <MediaIndicator script={latin} service="news">
+    <StyledTime datetime="PT2M15S">2:15</StyledTime>
```
Comment (sareh) [inline packages/components/psammead-story-promo/src/index.test.jsx:85]: Updating to pass in a Time component into Media Indicator in our test.

**Review by DenisHdz — APPROVED** [review]
LGTM 👍

File: packages/components/psammead-story-promo/src/index.stories.jsx [hunk 5]
```diff
@@ -22,16 +24,31 @@ const buildImg = () => (
...
-      duration={displayImage && type !== 'photogallery' && '2:15'}
-      datetime="PT2M15S"
+      script={script}
       service={service}
       type={type}
-    />
+      isInline={mediaIndicatorIsInline}
+    >
+      {!mediaIndicatorIsInline && (
+        <StyledTime dateTime="PT2M15S">2:15</StyledTime>
```
Comment (sareh) [inline packages/components/psammead-story-promo/src/index.stories.jsx:46]: Updating to pass in a Time component to our Media Indicator for these stories.

File: packages/components/psammead-story-promo/src/index.jsx [hunk 6]
```diff
@@ -110,6 +134,9 @@ export const Headline = styled.h3`
   padding-bottom: ${GEL_SPACING};
   ${({ service }) => getSerifMedium(service)}
   ${({ script, promoType }) => script && headlineTypography(script)[promoType]}
+  ${({ promoHasImage }) =>
+    !promoHasImage &&
+    `display: inline;`} /* Needed for aligning Media Indicator with Headline */
```
Comment (sareh) [inline packages/components/psammead-story-promo/src/index.jsx:139]: If promo does not have an image, then the headline needs to be displayed inline with the media indicator.

File: packages/components/psammead-story-promo/src/TextStyles.jsx [hunk 7]
```diff
@@ -135,10 +135,11 @@ const TextGridItem = styled.div`
 
     ${({ promoType }) =>
       promoType === 'leading' && leadingPromoTimestampPadding}
-  
+
     ${({ displayImage }) =>
       !displayImage &&
-      `>div{ display:inline-block; padding: 0; vertical-align:initial; } `}
+      `>div{ display:inline-block; vertical-align:initial; }
+       & svg{ margin: 0; }`}
```
Comment (sareh) [inline packages/components/psammead-story-promo/src/TextStyles.jsx:142]: Resetting margin on the Media Icon svg to 0 when there is no image (e.g. the media indicator is inline)

File: packages/components/psammead-story-promo/src/IndexAlsos/index.jsx [hunk 8]
```diff
@@ -54,24 +48,23 @@ const StyledIndexAlsosLink = styled.a`
   &:visited {
     color: ${C_METAL};
   }
+
+  & svg { margin: 0; } /* Reset Media Indicator SVG margin */
```
Comment (sareh) [inline packages/components/psammead-story-promo/src/IndexAlsos/index.jsx:52]: Setting margin on the svg within an index also to 0.

**Review by mmcculloch — APPROVED** [review]

**Review by sareh — COMMENTED** [review]
Just adding notes to each change, for reference when Code Reviewing.

**Comment by PriyaKR** [comment]
Looks good to me.

**Pull request merged**